libraries/chain/include/eosio/chain/chain_config_helper.hpp

```diff
@@ -0,0 +1,186 @@
+#pragma once
+
+#include <stddef.h>                                   /* offsetof */
+#include <type_traits>                                /* is_base_of */
+
+#include <boost/hana/map.hpp>                         /* make_map */
+#include <boost/hana/pair.hpp>                        /* make_pair */
+#include <boost/hana/type.hpp>                        /* type_c */
+#include <boost/hana/find.hpp>                        /* find */
+#include <boost/preprocessor/seq/for_each.hpp>        /* BOOST_PP_SEQ_FOR_EACH */
+#include <boost/preprocessor/seq/for_each_i.hpp>      /* BOOST_PP_SEQ_FOR_EACH_I */
+#include <boost/preprocessor/seq/size.hpp>            /* BOOST_PP_SEQ_SIZE */
+#include <boost/preprocessor/seq/elem.hpp>            /* BOOST_PP_SEQ_ELEM */
+#include <boost/preprocessor/repetition/repeat.hpp>   /* BOOST_PP_REPEAT */
```

Comment:
Do we really need to add all these macros? Can't we just pack/unpack manually. Seems like a maintenance mess.

Reply:
@heifner tbh, we have a lot of metaprogramming & macro boiler plates in our code (e.g. fc reflect) and I was not so experienced with this stuff so I decided to train a bit to get more familiarity with this in future. I had plans to make some kind of FC_REFLECT_RANGE to make it generic but later I put this in config helper.
So since this code is not going to be reusable like FC_REFLECT we can put there just resolved macros.
I'm really 50/50 on this. Either works for me.

Reply:
The contract also has to do something similar, right? I assume we are not going to have all of this for the contract code.

Reply:
@heifner CDT&contracts part is not implemented yet but you can check the unit tests:
unittests/test-contracts/params_test/params_test.cpp

Reply:
I think I'm in agreement with Kevin here. While a bit tedious, maintaining manual pack/unpack would likely be easier than trying to maintain moderately complex macros.

Reply:
If you want to make it easy to add the manual code, you could probably run just the preprocessor step and then copy/paste that to help you avoid some typing.

Reply:
@heifner @jeffreyssmith2nd please check my recent commit